app/models/user.rb

```diff
@@ -1,11 +1,38 @@
 class User < ApplicationRecord
+
+  validates :slug, uniqueness: true
+
+  def create_slug
+    #index = self.full_name.index(' ')
+   # self.full_name.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/,'')[index+2..-1]
+  
```

Comment:
* It is better to use split method
* the way you are doing this doesn't look safe. Seems like it may depend from case to case.